src/provision.toit

```diff
@@ -0,0 +1,508 @@
+  
+  scan_status -> ByteArray:
+    buffer := bytes.Buffer
+    w := protobuf.Writer buffer
+
+    resp_msg := {
+        MSG: MSG_RESP_STATUS,
+        RESP_STATUS: Map
+    }
+
+    if scan_done == false:
+      resp_msg[RESP_STATUS][RESP_STATUS_FINISHED] = 0
+    else:
+      resp_msg[RESP_STATUS][RESP_STATUS_FINISHED] = 1
+      resp_msg[RESP_STATUS][RESP_STATUS_COUNT] = ap_list.size
+
+    return protobuf_map_to_bytes_ --message=resp_msg
+  
+  scan_result r/protobuf.Reader -> ByteArray:
+    r.read_message:
+        r.read_field REQ_RESULT_COUNT:
+          report_count = r.read_primitive protobuf.PROTOBUF_TYPE_INT32
+
+    ap_info_msg := List
+    if msg_offset < ap_list.size:
+      ap_num := min (ap_list.size - msg_offset) report_count
+      ap_num.repeat:
+        ap ::= ap_list[msg_offset + it]
+        ap_info := {
+          RESP_RESULT_ENTRIES_SSID: ap.ssid,
+          RESP_RESULT_ENTRIES_CHANNEL: ap.channel,
+          RESP_RESULT_ENTRIES_RSSI: ap.rssi,
+          RESP_RESULT_ENTRIES_BSSID: ap.bssid,
+          RESP_RESULT_ENTRIES_AUTH: ap.authmode,
+        }
+        ap_info_msg.add ap_info
+
+      msg_offset += ap_num
+
+    resp_msg := {
+      MSG: MSG_RESP_RESULT,
+      RESP_RESULT: {
+        RESP_RESULT_ENTRIES: ap_info_msg
+      }
+    }
+    return protobuf_map_to_bytes_ --message=resp_msg
+
+  run data/ByteArray -> ByteArray:
+    resp := #[]
+    
+    r := protobuf.Reader data
+    r.read_message:
+      r.read_field MSG:
+        msgid := r.read_primitive protobuf.PROTOBUF_TYPE_INT32
+        if msgid == 2:
+          resp = scan_status
+      r.read_field REQ_START:
+        resp = scan_start r
+      r.read_field REQ_STATUS:
+        r.read_primitive protobuf.PROTOBUF_TYPE_BYTES
+        resp = scan_status
+      r.read_field REQ_RESULT:
+        resp = scan_result r
+
+    return resp
+
+class ConfigProcess_ implements Process_:
+  static MSG ::= 1 /** type: enum */
+
+  /** Message enum number */
+  static MSG_REQ_STATUS ::= 0
+  static MSG_RESP_STATUS ::= 1
+  static MSG_SET_CONFIG ::= 2
+  static MSG_RESP_CONFIG ::= 3
+  static MSG_SET_APPLY ::= 4
+  static MSG_RESP_APPY ::= 5
+
+  static REQ_STATUS ::= 10 /** type: message */
+
+  static RESP_STATUS ::= 11 /** type: message */
+  static RESP_STATUS_CONNECTED ::= 11 /** type: message */
+  static RESP_STATUS_CONNECTED_IPV4_ADDR ::= 1 /** type: string */
+  static RESP_STATUS_CONNECTED_AUTH_MODE ::= 2 /** type: uint32 */
+  static RESP_STATUS_CONNECTED_SSID ::= 3 /** type: bytes */
+  static RESP_STATUS_CONNECTED_BSSID ::= 4 /** type: bytes */
+  static RESP_STATUS_CONNECTED_CHANNEL ::= 5 /** type: uint32 */
+
+  static SET_CONFIG ::= 12 /** type: message */
+  static SET_CONFIG_SSID ::= 1 /** type: bytes */
+  static SET_CONFIG_PASSWORD ::= 2 /** type: bytes */
+
+  static REQ_APPLY ::= 14 /** type: message */
+
+  ssid/string := ""
+  password/string := ""
+  network := null
+
+  latch/monitor.Latch
+
+  constructor .latch/monitor.Latch:
```

Comment:
If a constructor parameter is directly used to initialize a field using `.`, then the type is automatically "copied" to the parameter.
```suggestion
  constructor .latch:
```